src/libstore/dummy-store.cc

```diff
@@ -61,15 +66,111 @@ struct DummyStore : public virtual DummyStoreConfig, public virtual Store
         RepairFlag repair, CheckSigsFlag checkSigs) override
     { unsupported("addToStore"); }
 
-    virtual StorePath addToStoreFromDump(
-        Source & dump,
+    StorePath addToStoreFromDump(
+        Source & source0,
         std::string_view name,
         FileSerialisationMethod dumpMethod = FileSerialisationMethod::Recursive,
         ContentAddressMethod hashMethod = FileIngestionMethod::Recursive,
         HashAlgorithm hashAlgo = HashAlgorithm::SHA256,
         const StorePathSet & references = StorePathSet(),
         RepairFlag repair = NoRepair) override
-    { unsupported("addToStore"); }
```

Comment:
I never liked the "dummy" store.
Should it be an always-failing store? A store that only pretends to write? An in memory store that can read what was written in the same process? Something for users or not? Guaranteed to work like that into the future?
Nothing prevents users from using it, so we should probably assume the worst and add new stores instead of changing this one, but maybe @Ericson2314 or @edolstra disagree?

Reply:
@roberth I think one can use read only mode to get back the old behavior. Can do a per store read only mode with default yes in dummy too.

Maybe it should be a unstable just for tests thing?

I don't want to do breakage, just don't wanna litter stable on off testing stores everywhere because we forgot to mark unstable either.

Reply:
Perhaps we could make `dummy` an alias for something like `transient://?read-only=true` then, and deprecate `dummy`?

Reply:
That sounds good, but I think we can just do later?

Reply:
(from friday)
- "memory://" allows reads and writes (eg: sqlite's ":memory:")
- "null://" writes succeed, but they do nothing
- "dummy://" makes writes and reads fail